src/stores/pacbioPools.js

```diff
@@ -46,8 +47,13 @@ export const usePacbioPoolsStore = defineStore('pacbioPools', {
     poolsArray: (state) => {
       return Object.values(state.pools).map((pool) => {
         const used_aliquots = pool.used_aliquots.map((used_aliquotId) => {
-          const { id, type, source_id, tag, run_suitability } = state.used_aliquots[used_aliquotId]
-          const { sample_name } = state.requests[source_id] || {}
+          const { id, type, source_id, source_type, tag, run_suitability } =
+            state.used_aliquots[used_aliquotId]
+          // Get the sample name based on the source_type
```

Comment:
I have assumed for libraries they want the libraries sample_name rather than just the library barcode.